correlation_context/HTTP_HEADER_FORMAT.md

```diff
@@ -10,24 +10,64 @@ A correlation context header is used to pass the name-value context properties f
 
 `Correlation-Context`
 
-Multiple correlation context headers are allowed. Values can be combined in a single header according to the [RFC 7230](https://tools.ietf.org/html/rfc7230#page-24).
-
 ## Header value
 
-`name1=value1[;properties1],name2=value2[;properties2]`
+The header value contains a list of key/value pairs. The first pair MUST contain the version of the header
+as `v=<version>`.
+
+The header MAY be sent or received as multiple header fields. Multiple header fields MUST be handled as specified by <a data-cite='!RFC7230#field.order'>RFC7230 Section 3.2.2 Field Order</a>. The header SHOULD be sent as a single field when possible, but MAY be split into multiple header fields. When sending the header as multiple header fields, it MUST be split according to <a data-cite='!RFC7230#field.order'>RFC7230</a>. When receiving multiple header fields, they MUST be combined into a single header according to <a data-cite='!RFC7230#field.order'>RFC7230</a>.
+
+When multiple headers are combined as described before, the resulting combined header will contain
+a version identifier for each individual header that was concatinated. This is intended and parsers have to expect that after the first version identifier another version identifier might occur at any time marking subsequent key/value pairs compliant with the denoted version until another version identifier occurs.
+
+
+This section uses the Augmented Backus-Naur Form (ABNF) notation of [[!RFC5234]], including the DIGIT rule in <a data-cite='!RFC5234#appendix-B.1'>appendix B.1 for RFC5234</a>. It also includes the `OWS` rule from <a data-cite='!RFC7230#whitespace'>RFC7230 section 3.2.3</a>.
+
+The `DIGIT` rule defines numbers `0`-`9`.
+
+The `OWS` rule defines an optional whitespace character. To improve readability, it is used where zero or more whitespace characters might appear.
+
+The caller SHOULD generate the optional whitespace as a single space; otherwise, a caller SHOULD NOT generate optional whitespace. See details in the <a data-cite='!RFC7230#whitespace'>corresponding RFC</a>.
+
+The field value is a `list` of `list-members` separated by commas (`,`). A `list-member` is a key/value pair separated by an equals sign (`=`). Spaces and horizontal tabs surrounding `list-member`s are ignored. There can be a maximum of 180 `list-member`s in a `list`.
+One single key/value pair mut not be larger than `4096 Bytes`.
+The total size of the header MUST not exceed `8192 Bytes`.
+
+### list
+
+A simple example of the header value might look like:
+
+`v=0,key1=opaqueValue1[;properties1],key2=opaqueValue2[;properties2]`
+
+@TODO: Provide ABNF once agreed upon
 
 **Limits:**
 1. Maximum number of name-value pairs: `180`.
 2. Maximum number of bytes per a single name-value pair: `4096`.
 3. Maximum total length of all name-value pairs: `8192`.
 
-## Name format
+## Version
+A key value pair `v=<version>` MUST be the first element of the list.
```

Comment:
This is potentially confusing for combined headers. It is not clear that given two headers, you mean first for each of them.